From b8249661fdeb04f51adedde2a526061e3b0fee4c Mon Sep 17 00:00:00 2001
From: xianming wang <mingwax@codeaurora.org>
Date: Thu, 25 Oct 2018 15:38:35 +0800
Subject: [PATCH 3/9] SnapdragonCamera: SunSet and Landscape use HAL-ZSL mode

SunSet and Landscape use HAL-ZSL mode as default.

CRs-Fixed: 2358000
Change-Id: I60e03ca3ec084c3fa85c3504c73fd54daf9f6279
---
 assets/dependency.json                      | 6 ++++--
 src/com/android/camera/SettingsManager.java | 6 +-----
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/assets/dependency.json b/assets/dependency.json
index 2a6339083..3ded5afb5 100644
--- a/assets/dependency.json
+++ b/assets/dependency.json
@@ -49,7 +49,8 @@
       "pref_camera2_longshot_key":"off"}
     ,
     "4":
-    {"pref_camera2_coloreffect_key":"0"}
+    {"pref_camera2_coloreffect_key":"0",
+     "pref_camera2_zsl_key":"hal-zsl"}
     ,
     "5":
     {"pref_camera2_coloreffect_key":"0",
@@ -62,7 +63,8 @@
     {"pref_camera2_coloreffect_key":"0"}
     ,
     "10":
-    {"pref_camera2_coloreffect_key":"0"}
+    {"pref_camera2_coloreffect_key":"0",
+      "pref_camera2_zsl_key":"hal-zsl"}
     ,
     "13":
     {"pref_camera2_coloreffect_key":"0"}
diff --git a/src/com/android/camera/SettingsManager.java b/src/com/android/camera/SettingsManager.java
index 517046574..dd798bc68 100755
--- a/src/com/android/camera/SettingsManager.java
+++ b/src/com/android/camera/SettingsManager.java
@@ -1786,11 +1786,7 @@ public class SettingsManager implements ListMenu.SettingsListener {
     public boolean isZSLInAppEnabled(){
         String value = getValue(KEY_ZSL);
         String appZSLValue = mContext.getString(R.string.pref_camera2_zsl_entryvalue_app_zsl);
-        if ( (value != null && value.equals(appZSLValue)) ||
-                SettingsManager.SCENE_MODE_SUNSET_STRING.equals(
-                        SettingsManager.getInstance().getValue(SettingsManager.KEY_SCENE_MODE)) ||
-                SettingsManager.SCENE_MODE_LANDSCAPE_STRING.equals(
-                        SettingsManager.getInstance().getValue(SettingsManager.KEY_SCENE_MODE))){
+        if ( value != null && value.equals(appZSLValue)){
             return true;
         }else{
             return false;
-- 
2.17.1

